### PR TITLE
[Fixes GEOS-8701] - Importer rest interface does not configure time dimension when a DateFormatTransform is defined on tasks

### PR DESCRIPTION
**Pull request by afabiani**
Replaces #2844

### CODE REVIEW

**Comment by afabiani** [comment]
@simboss @jodygarnett can you please take a look?

**Comment by afabiani** [comment]
Still getting odd and random errors from travis-push build. Mostly 90% of runs fails due to filesystem issues like the one below... trying to hit the build again!

image

**Comment by aaime** [comment]
Last build is fine.

File: src/extension/importer/core/src/test/java/org/geoserver/importer/ImportTransformTest.java [hunk 1]
```diff
@@ -166,7 +166,7 @@ public void testDateFormatTransform() throws Exception {
         context.setTargetStore(store);
 
         ImportTask task = context.getTasks().get(0);
-        task.getTransform().add(new DateFormatTransform("timestamp", "yyyy-MM-dd HH:mm:ss.S"));
+        task.getTransform().add(new DateFormatTransform("timestamp", "yyyy-MM-dd HH:mm:ss.S", null, null));
```
Comment (tbarsballe) [inline src/extension/importer/core/src/test/java/org/geoserver/importer/ImportTransformTest.java]: With the overloaded constructor, this change is no longer necessary

File: src/extension/importer/core/src/test/java/org/geoserver/importer/transform/TransformTest.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
     @Test
     public void testDateFormatTransform() throws Exception {
-        SimpleFeature f = transform(new DateFormatTransform("date", null),
+        SimpleFeature f = transform(new DateFormatTransform("date", null, null, null),
```
Comment (tbarsballe) [inline src/extension/importer/core/src/test/java/org/geoserver/importer/transform/TransformTest.java]: With the overloaded constructor, this change is no longer necessary

File: src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskAdvancedPage.java [hunk 3]
```diff
@@ -208,7 +208,7 @@ protected void onSubmit(AjaxRequestTarget target, Form<?> form) {
                                 if (dateFormat == null || "".equals(dateFormat.trim())) {
                                     dateFormat = null;
                                 }
-                                item.setModelObject(new DateFormatTransform(field, dateFormat));
+                                item.setModelObject(new DateFormatTransform(field, dateFormat, null, null));
```
Comment (tbarsballe) [inline src/extension/importer/web/src/main/java/org/geoserver/importer/web/ImportTaskAdvancedPage.java]: With the overloaded constructor, this change is no longer necessary

**Comment by tbarsballe** [comment]
Thanks, docs look good.

**Pull request merged**